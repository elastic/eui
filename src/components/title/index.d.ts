import { CommonProps } from '../common';
import { SFC } from 'react';

declare module '@elastic/eui' {
  /**
   * title type defs
   *
   * @see './title.js'
   */

  type EuiTitleSize = 'xxxs' | 'xxs' | 'xs' | 's' | 'm' | 'l';

  type EuiTitleTextTransform = 'uppercase';

  type EuiTitleProps = CommonProps & {
    size?: EuiTitleSize;
    textTransform?: EuiTitleTextTransform;
  };

  export const EuiTitle: SFC<EuiTitleProps>;
}
