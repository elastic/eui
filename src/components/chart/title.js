import {
  cloneElement,
} from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const titleSizeToClassNameMap = {
  s: 'euiTitle--small',
  l: 'euiTitle--large',
};

export const TITLE_SIZES = Object.keys(titleSizeToClassNameMap);

export const EuiTitle = ({ size, children, className, ...rest }) => {

  const classes = classNames(
    'euiTitle',
    titleSizeToClassNameMap[size],
    className
  );

  const props = {
    className: classes,
    ...rest
  };

  return cloneElement(children, props);
};

EuiTitle.propTypes = {
  children: PropTypes.element.isRequired,
  className: PropTypes.string,
  size: PropTypes.oneOf(TITLE_SIZES),
};
