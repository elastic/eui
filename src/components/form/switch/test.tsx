import React from 'react';

import { EuiSwitch } from './switch';

export const Test = ({ checked }: { checked: boolean }) => (
  <EuiSwitch
    checked={checked}
    label="Test"
    onChange={e => console.log(e.target.checked)}
  />
);
