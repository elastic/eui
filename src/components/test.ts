export function add(x: number, y: number) {
  return x + y;
}

add(5, 4);
