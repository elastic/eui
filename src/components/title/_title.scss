.euiTitle {
  + .euiTitle {
    margin-top: $euiSizeL;
  }
}

.euiTitle--uppercase {
  text-transform: uppercase;
}

.euiTitle--xxxsmall {
  @include euiTitle('xxxs');
}

.euiTitle--xxsmall {
  @include euiTitle('xxs');
}

.euiTitle--xsmall {
  @include euiTitle('xs');
}

.euiTitle--small {
  @include euiTitle('s');
}

.euiTitle--medium {
  @include euiTitle('m');
}

.euiTitle--large {
  @include euiTitle('l');
}
