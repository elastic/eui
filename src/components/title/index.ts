export { EuiTitle, EuiTitleProps, EuiTitleSize } from './title';
