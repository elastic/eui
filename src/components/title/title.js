import {
  cloneElement,
} from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

const titleSizeToClassNameMap = {
  xxxs: 'euiTitle--xxxsmall',
  xxs: 'euiTitle--xxsmall',
  xs: 'euiTitle--xsmall',
  s: 'euiTitle--small',
  m: 'euiTitle--medium',
  l: 'euiTitle--large',
};

export const TITLE_SIZES = Object.keys(titleSizeToClassNameMap);

const textTransformToClassNameMap = {
  uppercase: 'euiTitle--uppercase',
};

export const TEXT_TRANSFORM = Object.keys(textTransformToClassNameMap);

export const EuiTitle = ({ size, children, className, textTransform, ...rest }) => {

  const classes = classNames(
    'euiTitle',
    titleSizeToClassNameMap[size],
    textTransformToClassNameMap[textTransform],
    className
  );

  const props = {
    className: classes,
    ...rest
  };

  return cloneElement(children, props);
};

EuiTitle.propTypes = {
  children: PropTypes.element.isRequired,
  className: PropTypes.string,
  size: PropTypes.oneOf(TITLE_SIZES).isRequired,
  textTransform: PropTypes.oneOf(TEXT_TRANSFORM),
};

EuiTitle.defaultProps = {
  size: 'm',
};
