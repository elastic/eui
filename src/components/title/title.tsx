/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0 and the Server Side Public License, v 1; you may not use this file except
 * in compliance with, at your election, the Elastic License 2.0 or the Server
 * Side Public License, v 1.
 */

import React, { FunctionComponent, ReactElement } from 'react';
import classNames from 'classnames';
import { CommonProps, keysOf } from '../common';

const titleSizeToClassNameMap = {
  xxxs: 'euiTitle--xxxsmall',
  xxs: 'euiTitle--xxsmall',
  xs: 'euiTitle--xsmall',
  s: 'euiTitle--small',
  m: 'euiTitle--medium',
  l: 'euiTitle--large',
};

export const TITLE_SIZES = keysOf(titleSizeToClassNameMap);
export type EuiTitleSize = keyof typeof titleSizeToClassNameMap;

const textTransformToClassNameMap = {
  uppercase: 'euiTitle--uppercase',
};

export const TEXT_TRANSFORM = keysOf(textTransformToClassNameMap);
export type EuiTitleTextTransform = keyof typeof textTransformToClassNameMap;

export type EuiTitleProps = CommonProps & {
  /**
   * ReactElement to render as this component's content
   */
  children: ReactElement<any>;
  size?: EuiTitleSize;
  textTransform?: EuiTitleTextTransform;
  id?: string;
};

export const EuiTitle: FunctionComponent<EuiTitleProps> = ({
  size = 'm',
  children,
  className,
  textTransform,
  ...rest
}) => {
  const classes = classNames(
    'euiTitle',
    titleSizeToClassNameMap[size],
    textTransform ? textTransformToClassNameMap[textTransform] : undefined,
    className,
    children.props.className
  );

  const props = {
    className: classes,
    ...rest,
  };

  return React.cloneElement(children, props);
};
