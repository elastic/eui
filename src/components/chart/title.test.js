import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiTitle } from './title';

describe('EuiTitle', () => {
  test('is rendered', () => {
    const component = render(
      <EuiTitle {...requiredProps}>
        <h1>Title</h1>
      </EuiTitle>
    );

    expect(component)
      .toMatchSnapshot();
  });
});
