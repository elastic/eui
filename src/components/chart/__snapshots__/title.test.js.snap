// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiTitle is rendered 1`] = `
<h1
  aria-label="aria-label"
  class="euiTitle testClass1 testClass2"
  data-test-subj="test subject string"
>
  Title
</h1>
`;
